### PR TITLE
1.2 is done

### PR DESCRIPTION
**Pull request by dmytro-hryhoruk**

### CODE REVIEW

File: src/main/java/core/basesyntax/BooleanExpression.java [hunk 1]
```diff
@@ -13,6 +13,13 @@
      * <p>Это задание необходимо решать с использованием логических операторов.</p>
      */
     public boolean booleanExpression(boolean a, boolean b, boolean c, boolean d) {
+        int booleanCounter =0;
+        if(a)booleanCounter++;
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/BooleanExpression.java]: Read task carefully! We should use boolean operators without `if` statement

**Review by Sofasmile — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/BooleanExpression.java [hunk 2]
```diff
@@ -13,6 +13,12 @@
...
      */
     public boolean booleanExpression(boolean a, boolean b, boolean c, boolean d) {
-        return false;
+        return ((a && b) && (!c && !d))
+                || ((a && c) && (!b && !d))
+                || ((a && d) && (!b && !c))
+                || ((b && c) && (!a && !d))
+                || ((c && d) && (!a && !b))
+                || ((b & d) && (!c && !a))
+                ? true : false;
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/BooleanExpression.java]: do we need it `? true : false`?

File: src/main/java/core/basesyntax/BooleanExpression.java [hunk 3]
```diff
@@ -13,6 +13,12 @@
      * <p>Это задание необходимо решать с использованием логических операторов.</p>
      */
     public boolean booleanExpression(boolean a, boolean b, boolean c, boolean d) {
-        return false;
+        return ((a && b) && (!c && !d))
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/BooleanExpression.java]: good job, but try to simplify

**Review by Sofasmile — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/BooleanExpression.java (hunk 2 above)
Comment (dmytro-hryhoruk) [reply src/main/java/core/basesyntax/BooleanExpression.java]: No, actually

**Review by Sofasmile — APPROVED** [review]
God work! But try understanding how it work

**Pull request closed without merging**